src/MessageBox.Avalonia/MessageBoxManager.cs

```diff
@@ -19,32 +18,41 @@ public static IMsBoxWindow<string> GetMessageBoxCustomWindow(MessageBoxCustomPar
             var window = new CustomWindow(@params.Style);
             @params.Window = window;
             window.DataContext = new MsBoxCustomViewModel(@params);
-            return new MsBoxCustomWindow(window);
+            return new MsBoxWindowBase<CustomWindow, string>(window, window.ButtonResult);
         }
 
         public static IMsBoxWindow<ButtonResult> GetMessageBoxStandardWindow(MessageBoxStandardParams @params)
         {
             var window = new StandardWindow(@params.Style);
             @params.Window = window;
             window.DataContext = new MsBoxStandardViewModel(@params);
-            return new MsBoxStandardWindow(window);
+            return new MsBoxWindowBase<StandardWindow,ButtonResult>(window, window.ButtonResult);
+        }
+        public static IMsBoxWindow<ButtonResult> GetMessageBoxHyperlinkWindow(MessageBoxHyperlinkParams @params)
+        {
+            var window = new HyperlinkWindow(@params.Style) { DataContext = new MsBoxHyperlinkViewModel(@params) };
+            @params.Window = window;
+            return new  MsBoxWindowBase<HyperlinkWindow, ButtonResult>(window, window.ButtonResult);
         }
-
         public static IMsBoxWindow<ButtonResult> GetMessageBoxStandardWindow(string title, string text,
             ButtonEnum @enum = ButtonEnum.Ok, Icon icon = Icon.None,
             WindowStartupLocation windowStartupLocation = WindowStartupLocation.CenterScreen,
             Style style = Style.None) => GetMessageBoxStandardWindow(new MessageBoxStandardParams
-        {
-            ContentTitle = title, ContentMessage = text, ButtonDefinitions = @enum, Icon = icon, Style = style,
-            WindowStartupLocation = windowStartupLocation
-        });
+            {
```

Comment:
This going too far in indent ;)

Reply:
@rstm-sf lol previously you said what its individual

Reply:
That's right, you walked away from the canon